tests/h/views/admin_groups_test.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import datetime
+
+import pytest
+import mock
+
+# from h.models.auth_client import AuthClient, GrantType, ResponseType
+from h.views import admin_groups
+from h.views.admin_groups import GroupCreateController
+
+
+def test_index_lists_groups_sorted_by_created_desc(pyramid_request, routes, factories, authority):
+    groups = [factories.Group(created=datetime.datetime(2017, 8, 2)),
+              factories.Group(created=datetime.datetime(2015, 2, 1)),
+              factories.Group(),
+              factories.Group(created=datetime.datetime(2013, 2, 1))]
+
+    ctx = admin_groups.groups_index(None, pyramid_request)
+
+    # We can't avoid getting the Public group back, which is created outside of
+    # these tests' sphere of influence. Remove it as it is not feasible to
+    # assert where it will appear in creation order.
+    filtered_groups = list(filter(lambda group: group.pubid != u'__world__',
+                                  ctx['results']))
+
+    expected_groups = [groups[2], groups[0], groups[1], groups[3]]
+    assert filtered_groups == expected_groups
+
+
+def test_index_paginates_results(pyramid_request, routes, paginate):
+    admin_groups.groups_index(None, pyramid_request)
+
+    paginate.assert_called_once_with(pyramid_request, mock.ANY, mock.ANY)
+
+
+class TestGroupCreateController(object):
+
+    def test_get_sets_form(self, pyramid_request):
+        ctrl = GroupCreateController(pyramid_request)
+
+        ctx = ctrl.get()
+
+        assert 'form' in ctx
+
+    def test_it_handles_form_submission(self, pyramid_request, handle_form_submission, matchers):
```

Comment:
Because of our layers of abstraction and mocking, it is nearly impossible to test forms in any meaningful way without contortions, so I concede that at present this view is under-tested. After a discussion with @seanh earlier, I've elected to put this aside for the moment so that I can get something out the door first.